src/main/java/org/opentripplanner/middleware/utils/ItineraryExistenceChecker.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.OtpDispatcher;
+import org.opentripplanner.middleware.otp.OtpDispatcherResponse;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.OtpResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * This utility class checks for existence of itineraries for given OTP queries.
+ */
+public class ItineraryExistenceChecker {
+    private final Function<String, OtpDispatcherResponse> otpDispatcherFunction;
+
+    /**
+     * @param otpDispatcherFunction a function that takes String as input and returns {@link OtpDispatcherResponse},
+     *                              example: {@link OtpDispatcher#sendOtpPlanRequest}.
+     */
+    public ItineraryExistenceChecker(Function<String, OtpDispatcherResponse> otpDispatcherFunction) {
+        if (otpDispatcherFunction == null) throw new NullPointerException();
+        this.otpDispatcherFunction = otpDispatcherFunction;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @param labeledQueries a map containing the queries to check, each query having a key or
+     *                       label that will be used in Result.labeledResponses for easy identification.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public Result checkAll(Map<String, String> labeledQueries, boolean checkArrival) {
```

Comment:
I'd recommend renaming `chechArrival` to `tripIsArriveBy`.

Reply:
Good suggestion.